tools/reactive-controllers/test/pending-state.test.ts

```diff
@@ -0,0 +1,153 @@
+    });
+
+    afterEach(() => {
+        sinon.restore();
+    });
+
+    describe('renderPendingState', () => {
+        it('should change aria-label of host when pending and when not pending', async () => {
+            await host.updateComplete;
+            let ariaLabel = host.getAttribute('aria-label');
+            expect(ariaLabel).to.equal('Pending');
+
+            host.removeAttribute('pending');
+            await host.updateComplete;
+            ariaLabel = host.getAttribute('aria-label');
+            expect(ariaLabel).to.equal('clickable');
+
+            host.setAttribute('pending', 'true');
+            await host.updateComplete;
+            ariaLabel = host.getAttribute('aria-label');
+            expect(ariaLabel).to.equal('Pending');
+        });
+
+        it('should render the pending state UI', async () => {
+            const pendingLabel = 'Custom Pending Label';
+            host.pendingLabel = pendingLabel;
+            const templateResult = controller.renderPendingState();
+
+            const renderedElement = await fixture(html`
+                ${templateResult}
+            `);
+            const expectedElement = await fixture(html`
+                <sp-progress-circle
+                    id="loader"
+                    size="s"
+                    indeterminate
+                    aria-valuetext=${pendingLabel}
+                    class="progress-circle"
+                ></sp-progress-circle>
+            `);
+
+            expect(renderedElement.outerHTML === expectedElement.outerHTML).to
+                .be.true;
+        });
+
+        it('should render the default pending state UI if no label is provided', async () => {
+            host.pendingLabel = undefined;
+            const templateResult = controller.renderPendingState();
+            const renderedElement = await fixture(html`
+                ${templateResult}
+            `);
+            const expectedElement = await fixture(html`
+                <sp-progress-circle
+                    id="loader"
+                    size="s"
+                    indeterminate
+                    aria-valuetext="Pending"
+                    class="progress-circle"
+                ></sp-progress-circle>
+            `);
+
+            const renderedAttributes = renderedElement.attributes;
+            const expectedAttributes = expectedElement.attributes;
+
+            expect(renderedAttributes.length === expectedAttributes.length).to
+                .be.true;
+
+            for (let i = 0; i < renderedAttributes.length; i++) {
+                const renderedAttr = renderedAttributes[i];
+                const expectedAttr = expectedAttributes.getNamedItem(
+                    renderedAttr.name
+                );
+
+                expect(renderedAttr.value === expectedAttr?.value).to.be.true;
+            }
+        });
+
+        it('should toggle the pending state on and off and preserve the component state correctly', async () => {
```

Comment:
i see we are checking for the progress circle state, but should to also check the host state for no side-effects?